src/conn.c

```diff
@@ -2139,9 +2139,14 @@ _evStopPolling(natsConnection *nc)
 static bool
 _processOpError(natsConnection *nc, natsStatus s, bool initialConnect)
 {
+    bool forceReconnect;
+
     natsConn_Lock(nc);
 
-    if (!initialConnect)
+    forceReconnect = initialConnect || nc->forceReconnect;
```

Comment:
The `go` client disregards/overrides its equivalent of `nc->opts->allowReconnect` when `Disconnect()` is invoked, I posed a question about it in https://github.com/nats-io/nats.go/pull/1624#issuecomment-2091859745.

If we were to error out when not `nc->opts->allowReconnect` - no changes would be needed, otherwise I need to override it (present implementation).

Reply:
Ok, but that also means that the user would not be able to reconnect only "manually", the connection will auto-reconnect on all relevant errors.

I'll make the change now.

Reply:
Sorry I don't understand. The original check here says to check for the current connection status so that we don't attempt to reconnect if we are in the middle of it. The check for `!initialConnect` was likely introduced when adding the ability to connect asynchronously (that is, call `natsConnection_Create()` (or Connect) while a server is not running does not fail, but attempt the reconnect right away.

Reply:
I just meant that one won't be able to make a connection with allowReconnect==false if so desired, but then forcefully `Reconnect()` on it (say if they want to re-auth?). That was my understanding of the use-case.

Reply:
That's why maybe don't do the changes until it is agreed with the rest of the team. I did not mean to have you make all the changes if you have to put it back because this is how the team wants the feature to work.

If as a team it is agreed that it should override, then simply add to the doc that this function ignores the "allow reconnect" option (currently it says that it respects them all).

Reply:
(will not merge until verified)

Reply:
See the comment from the team feedback: ignore `allowReconnect` altogether, just close the socket and let the client do what it does.